PlaySpotApi/Routes/LocationRoutes.cs

```diff
@@ -55,6 +59,74 @@ public static RouteGroupBuilder MapLocationRoutes(this RouteGroupBuilder group)
             .Produces<List<Location>>(StatusCodes.Status200OK)
             .Produces(StatusCodes.Status404NotFound);
 
+
+            //this get request returns all the locations for a given sport with a specified radius.
+            group.MapGet("/locations-by-sport-", async (
+                string sportName,
+                string coordinates,   
+                double radius,        // in km
+                PlaySpotDbContext db) =>
+            {
+                // 1) Parse the user's coordinates
+                var userParts = coordinates
+                    .Split(',', StringSplitOptions.RemoveEmptyEntries);
+                if (userParts.Length != 2
+                    || !double.TryParse(userParts[0], out var userLat)
+                    || !double.TryParse(userParts[1], out var userLon))
+                {
+                    return Results.BadRequest("Invalid 'coordinates' format. Use 'lat,lon'.");
+                }
+
+                // 2) Fetch all locations that offer the sport
+                var all = await db.Locations
+                    .Include(l => l.Sports)
+                    .Where(l => l.Sports
+                        .Any(s => s.Name
+                            .Equals(sportName, StringComparison.OrdinalIgnoreCase)))
+                    .Select(l => new
+                    {
+                        l.LocationId,
+                        l.Name,
+                        l.Address,
+                        l.Coordinates,   // still as "lat,lon"
+                        Sports = l.Sports
+                            .Select(s => new { s.SportId, s.Name })
+                            .ToList()
+                    })
+                    .ToListAsync();
+
+                // 3) Parse each location's coords, filter by radius
+                var withinRadius = new List<object>();
+                foreach (var loc in all)
+                {
+                    var parts = loc.Coordinates
```

Comment:
Can you adjust this so the warning (`Dereference of a possibly null reference.`) is gone